starboard/xb1/tools/xb1_launcher.py

```diff
@@ -425,33 +438,37 @@ def Deploy(self):
       raise IOError('Packaged appx not found in package directory. Perhaps '
                     'package_cobalt script did not complete successfully.')
 
-    existing_package = self.CheckPackageIsDeployed()
+    existing_package = self.CheckPackageIsDeployed(_DEFAULT_PACKAGE_NAME)
     if existing_package:
       self._LogLn('Existing YouTube app found on device. Uninstalling.')
       self.WinAppDeployCmd('uninstall -package ' + existing_package)
 
+    if not self.CheckPackageIsDeployed(_DEBUG_VC_LIBS_PACKAGE_NAME):
+      self._LogLn('Required dependency missing. Attempting to install.')
+      self.WinAppDeployCmd(f'install -file "{_DEBUG_VC_LIBS_PATH}"')
+
     self._LogLn('Deleting temporary files')
     self._network_api.ClearTempFiles()
 
     try:
-      self._LogLn('Installing appx file ' + appx_package_file)
-      self.WinAppDeployCmd('install -file ' + appx_package_file)
+      self.WinAppDeployCmd(f'install -file {appx_package_file}')
     except subprocess.CalledProcessError:
       # Install exited with non-zero status code, clear everything out, restart,
       # and attempt another install.
       self._LogLn('Error installing appx. Attempting a clean install...')
       self.UninstallSubPackages()
+      self.DeleteLooseApps()
       self.RestartDevkit()
-      self.WinAppDeployCmd('install -file ' + appx_package_file)
+      self.WinAppDeployCmd(f'install -file {appx_package_file}')
 
     # Cleanup starboard arguments file.
     self.InstallStarboardArgument(None)
 
   # Validate that app was installed correctly by checking to make sure
   # that the full package name can now be found.
-  def CheckPackageIsDeployed(self):
+  def CheckPackageIsDeployed(self, package_name):
     package_list = self.WinAppDeployCmd('list')
-    package_index = package_list.find(_DEFAULT_PACKAGE_NAME)
+    package_index = package_list.find(package_name)
     if package_index == -1:
       return False
     return package_list[package_index:].split('\n')[0].strip()
```

Comment:
nit: it seems like there's the "query" and "result" name but they can differ? Not important but a comment here stating an example would be helpful reference for future readers.

Reply:
Yeah, when it's installed there's more in the full app name than just 'GoogleInc.YouTube', but it changes with each build. We return the full name here to reference it for other commands like uninstall or launch.